Fix #6383: macOS deprecation errors when compiling

AaronVanGeffen opened the pull request:
Compiling on macOS High Sierra (10.13) yields deprecation errors for the `NSFileHandlingPanelCancelButton` and `NSFileHandlingPanelOKButton`. Instead, `NSModalResponseCancel` and `NSModalResponseOK` should be used, respectively.

The new constants were introduced in 10.9, so there should not be any compatibility issues.

Gymnasiast commented on src/openrct2-ui/UiContext.macOS.mm:
```diff
@@ -102,7 +102,7 @@ void ShowMessageBox(SDL_Window * window, const std::string &message) override
                 panel.allowedFileTypes = extensions;
                 panel.directoryURL = [NSURL fileURLWithPath:directory];
-                if ([panel runModal] == NSFileHandlingPanelCancelButton){
+                if ([panel runModal] == NSModalResponseCancel){
```
Could you fix the braces here?

Gymnasiast commented:
Could you stick it in the changelog as well? Put it as "Technical".

LRFLEW commented:
>The new constants were introduced in 10.9, so there should not be any compatibility issues.

Aren't we still targeting 10.8 as the minimum? Are the constants numerically the same, or does it change something in how the code works?

Gymnasiast commented:
Well, I was asking myself that, but wasn't sure.

Perhaps a good question to ask: do we _want_ to target 10.8? If so, we might need a ifdef. But perhaps we shouldn't - after all, Apple dropped support for it in 2015.

LRFLEW commented:
>Perhaps a good question to ask: do we want to target 10.8?

I forget when/where we had this discussion, but I seem to recall that we decided to fully support the last 3 versions of macOS, and provide "best effort" for everything else we could. I would say using a deprecated constant would fall under "best effort", but that's just me 😜. Just know that increasing the minimum macOS version will cut of certain machines from running it, due to how Apple's releases work.

Gymnasiast commented:
Which machines? I seem to remember the supported machines staying the same for very long, until Sierra was released and dropped the 2009 MBPs, for example. But for those, El Capitan will still work.

AaronVanGeffen commented:
@LRFLEW Good to see you on top of these matters. Three versions ago is now 10.10 (Yosemite), though, which does support the new constants. Fortunately, even most 2008 hardware is still [supported by 10.10](https://en.wikipedia.org/wiki/OS_X_Yosemite#System_requirements), so I do not think there's any harm done by upping the requirements, here.

@Gymnasiast Thanks, I've addressed your comments per 129a01a86be6ed1772e05d8c85b5a923d12a9ccd

LRFLEW commented:
I went ahead and checked this out locally. Apparently, the deprecation is just for the name, and the values are identical. (In fact, the old one is assigned to the new one in its definition). Since we don't really support compiling in anything but the latest Xcode, I have no complaints with this change.
